### PR TITLE
feature(datatable): hide sort by select on mobile if not needed

### LINKED ISSUE #9209: [Bug Report] Data table headers replaced with "Sort by" selector on mobile screen
### Environment
**Vuetify Version:** 2.1.0
**Last working version:** 1.5.18
**Vue Version:** 2.6.10
**Browsers:** Chrome 77.0.3865.90
**OS:** Windows 10

### Steps to reproduce
Open Pen, activate development mode in Chrome and turn on mobile view (or open pen on mobile).

As long as the data table has custom item view implemented, headers are replaced with a Sort By selector (where does that come from?) even if sorting is disabled.

### Expected Behavior
Headers are visible.

### Actual Behavior
Headers are replaced with a Sort by selector.

### Reproduction Link
<a href="https://codepen.io/Tallyrald/pen/mddbVPv" target="_blank">https://codepen.io/Tallyrald/pen/mddbVPv</a>

### Other comments
It is also interesting to notice that rows are not displayed as "blocks" below each other (like how the examples do in the API docs), but stay as scrollable rows.

### PR DESCRIPTION
**Pull request by TimvdEijnden**
## Description
Hides the select element on mobile when there are no sortable headers

## Motivation and Context
fixes #9209

## How Has This Been Tested?
visually

## Markup:
https://jsfiddle.net/1s0xua3f/

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
I'm really not sure if this can be considered a fix because as of now, it is intended functionality. Maybe @nekosaur has a different opinon.

**Comment by nekosaur** [comment]
I am not against hiding the sort on mobile if there is nothing to sort. But @johnleider is correct that this is technically not a bug since the current behavior of disabling it is intentional.

I'd say this could probably go in a patch though, since functionality is not changed? 🤷‍♂

File: packages/vuetify/src/components/VDataTable/VDataTableHeaderMobile.ts [hunk 1]
```diff
@@ -43,17 +43,14 @@ export default mixins(header).extend({
         },
       }, children)
     },
-    genSortSelect () {
-      const sortHeaders = this.headers.filter(h => h.sortable !== false && h.value !== 'data-table-select')
-
+    genSortSelect (items: Array<any>) {
```
Comment (nekosaur) [inline packages/vuetify/src/components/VDataTable/VDataTableHeaderMobile.ts]: We use `any[]` syntax for array typings. It should also be `TableHeader[]`

Comment (TimvdEijnden) [reply packages/vuetify/src/components/VDataTable/VDataTableHeaderMobile.ts]: I'll change it. Do you also want me to use different commit msg and change the PR name not to include fix?

Comment (nekosaur) [reply packages/vuetify/src/components/VDataTable/VDataTableHeaderMobile.ts]: It should go to dev, not master

Comment (TimvdEijnden) [reply packages/vuetify/src/components/VDataTable/VDataTableHeaderMobile.ts]: Changed

**Comment by TimvdEijnden** [comment]
@nekosaur changed have been applies, would be great to move this forward if possible

**Pull request merged**